packages/driver/src/cy/commands/querying.js

```diff
@@ -486,11 +487,24 @@ module.exports = (Commands, Cypress, cy) => {
       // When multiple space characters are considered as a single whitespace in all tags except <pre>.
       const normalizeWhitespaces = (elem) => {
         let testText = elem.textContent || elem.innerText || $.text(elem)
+        const whitespaceStyle = $(elem).css('white-space')
 
-        if (elem.tagName === 'PRE') {
+        if (elem.tagName === 'PRE' ||
+        options.matchWhitespaces ||
+        whitespaceStyle === 'pre' ||
+        whitespaceStyle === 'pre-wrap' ||
+        whitespaceStyle === 'break-spaces') {
           return testText
         }
 
+        if (whitespaceStyle === 'pre-line') {
+          const lines = testText.replace(spaces, ' ').split('\n').map((line) => {
+            return line.trim()
+          })
+
+          return lines.join('\n')
+        }
```

Comment:
I think we can't add this behavior regarding the `white-space` CSS property without this being a breaking change. It should only take into account the `matchWhitespace` property. Maybe it makes sense to also handle `pre-line` specially, but again, only if `matchWhitespace: true`.

Reply:
I'm curious if everything will "just work", removing the need to check for `white-space`, or even having a `matchWhitespace` option, if the calls to `normalizeWhitespaces(elem)` are just replaced with `let testText = elem.innerText`. See [Node.textContent#Differences from innerText](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent#Differences_from_innerText) and [HTMLElement.innerText](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText), which states:

> The innerText property of the HTMLElement interface represents the "rendered" text content of a node and its descendants. **As a getter, it approximates the text the user would get if they highlighted the contents of the element with the cursor and then copied it to the clipboard.**

The current code for `normalizeWhitespaces` gives `textContent` higher precedent than `innerText`. I also found that Cypress currently includes the text for any inline styles in the `testText`, which is consistent with the documentation for `textContent.